modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -227,6 +227,19 @@ Currently, the following file formats are supported:
 */
 CV_EXPORTS_W Mat imread( const String& filename, int flags = IMREAD_COLOR );
 
+
+/** @brief Loads an image from a file.
+
+This is an overloaded member function, provided for convenience. It differs from the above function only in what argument(s) it accepts.
+@param filename Name of file to be loaded.
+@param img object in which the image will be loaded.
+@param flags Flag that can take values of cv::ImreadModes
+@note
+The image passing trought the img parameter can be pre-allocated. The memory is reused if the shape and the type match with the load image.
+ */
+CV_EXPORTS_W Mat imread( const String& filename, Mat & img, int flags = IMREAD_COLOR );
```

Comment:
As this method marked as `CV_EXPORTS_W` and wrapped to Python, please add a test to /home/dkurt/opencv/modules/python/test. This is useful for users to see how it works and control regressions.